internal/resources/fabric/connection/datasources.go

```diff
@@ -0,0 +1,201 @@
+package connection
+
+import (
+	"context"
+	"fmt"
+	"github.com/equinix/equinix-sdk-go/services/fabricv4"
+	"github.com/equinix/terraform-provider-equinix/internal/config"
+	"github.com/equinix/terraform-provider-equinix/internal/converters"
+	equinix_errors "github.com/equinix/terraform-provider-equinix/internal/errors"
+	equinix_schema "github.com/equinix/terraform-provider-equinix/internal/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"strings"
+)
+
+func DataSource() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceFabricConnectionRead,
+		Schema:      readFabricConnectionResourceSchema(),
+		Description: "Fabric V4 API compatible data resource that allow user to fetch connection for a given UUID",
+	}
+}
+
+func dataSourceFabricConnectionRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	uuid, _ := d.Get("uuid").(string)
+	d.SetId(uuid)
+	return resourceFabricConnectionRead(ctx, d, meta)
+}
+
+func DataSourceSearch() *schema.Resource {
```

Comment:
I'm a wary of having two datasources implemented in a single file, but that's mitigated somewhat by the fact that this is now an isolated package.